clang/lib/AST/DeclCXX.cpp

```diff
@@ -3423,6 +3425,30 @@ VarDecl *BindingDecl::getHoldingVar() const {
   return VD;
 }
 
+void DecompositionDecl::VisitHoldingVars(
```

Comment:
Rather than both of these functions, I'd prefer we have some sort of iterators that allow standard iteration over these.  We have 'filter' iterators already that take a lambda that we could define to allow going through these.

The visitation can be nice, but it has the requirement of going through every element even when we only care about 'the first of'.